S3Proxy/cache.go

```diff
@@ -5,45 +5,104 @@ import (
 	"time"
 )
 
-var s3Buckets = map[string]*s3BucketCacheItem{}
+var s3Buckets = map[string]*S3BucketCacheItem{}
+var s3Objects = map[string]*S3ObjectCacheItem{}
 
-type s3BucketCacheItem struct {
-	Name      string
-	Location  string
+type s3CacheItem struct {
```

Comment:
I think CacheItem maybe sounds better since it more generic. Also don't start the naming lowercase.